### ISSUE TITLE
`task export` gives non-deterministic task order

### ISSUE THREAD

**Issue by willhansen** [GothenburgBitFactory/taskwarrior#3527]
#### To report a bug...

* What command(s) did you run?
`diff -u <(task export) <(sleep 0.5;task export)`

* What did you expect to happen?
Deterministic output suitable for git tracking

* What actually happened?
The line order of the output is not deterministic, particularly for completed or deleted tasks with `id:0`
eg: 
`{"id":0,"description":"sleep","end":"20240628T000041Z","entry":"20240622T075543Z","modified":"20240628T000041Z","status":"completed","uuid":"dd12ffeb-3ccf-4f39-969a-7638d7eefd04","urgency":0.0438356},
{"id":0,"description":"asdf","end":"20240626T032134Z","entry":"20240614T044317Z","modified":"20240626T032137Z","status":"deleted","uuid":"ecc565bb-b102-4e27-b940-ce1018051f4b","urgency":0.0876712}`

* Paste the output of the ```task diag``` command.

```
task 3.0.2
   Platform: Linux

Compiler
    Version: 13.3.0
       Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
 Compliance: C++17

Build Features
      CMake: 3.29.3
    libuuid: libuuid + uuid_unparse_lower
 Build type: Release

Configuration
       File: /home/will/.config/task/taskrc (found), 52 bytes, mode 100644
       Data: /home/will/.task (found), dir, mode 40755
    Locking: Enabled
         GC: Enabled
    $EDITOR: nvim
Hooks
     System: Enabled
   Location: /home/will/.task/hooks
             (-none-)

Tests
   Terminal: 166x45
       Dups: Scanned 39 tasks for duplicate UUIDs:
             No duplicates found
 Broken ref: Scanned 39 tasks for broken references:
             No broken references found
```

**Comment by djmitche** [comment]
It looks like `task export` can actually take a report name, and will use the filter and sort from that report. That's not documented - maybe it should be?

It's a little hard to follow the code, but it looks like in the absence of a report it defaults to sorting by `id`, and I think that's what you're seeing -- the tasks with unique `id` are sorted numerically, but the tasks not in the working set all have `id` zero and the sort does not affect them.

If you have the capability to build locally, you could try changing `src/commands/CmdExport.cpp` where it defaults to `id` (`// if no sort order, sort by id`) to `id,uuid`.
